tests/test_parse.py

```diff
@@ -1,403 +1,404 @@
+        assert union_decl.cursor.is_anonymous()
+        assert union_decl.cursor.spelling == ""
+
+    def test_translation_unit(self):
+        self.__parse("")
+
+        tranlation_unit = self.tree_paths[0][0]
+        assert tranlation_unit.cursor.kind == clang.CursorKind.TRANSLATION_UNIT
+        assert tranlation_unit.cursor.spelling == self.file.name
+
+    def test_namespace(self):
+        self.__parse("namespace a_namespace {}")
+
+        namespace = self.tree_paths[0][1]
+        assert namespace.cursor.kind == clang.CursorKind.NAMESPACE
+        assert namespace.cursor.spelling == "a_namespace"
+
+    def test_namespace_ref(self):
+        self.__parse(
+            """
+        #include <ostream>
+        std::ostream anOstream;
+        """
+        )
+
+        inclusion_directive = self.tree_paths[0][1]
+        assert inclusion_directive.cursor.kind == clang.CursorKind.INCLUSION_DIRECTIVE
+        assert inclusion_directive.cursor.spelling == "ostream"
+
+        namespace_ref = self.tree_paths[1][2]
+        assert namespace_ref.cursor.kind == clang.CursorKind.NAMESPACE_REF
+        assert namespace_ref.cursor.spelling == "std"
+
+    def test_var_decl(self):
+        self.__parse("int anInt = 1;")
+
+        var_decl = self.tree_paths[0][1]
+        assert var_decl.cursor.kind == clang.CursorKind.VAR_DECL
+        assert var_decl.cursor.type.kind == clang.TypeKind.INT
+        assert var_decl.cursor.spelling == "anInt"
+
+    def test_field_decl(self):
+        self.__parse(
+            """
+        struct AStruct {
+            int aClassMember;
+        };
+        """
+        )
+
+        field_decl = self.tree_paths[0][2]
+        assert field_decl.cursor.kind == clang.CursorKind.FIELD_DECL
+        assert field_decl.cursor.type.kind == clang.TypeKind.INT
+        assert field_decl.cursor.spelling == "aClassMember"
+
+    def test_function_decl_without_parameters(self):
+        self.__parse(
+            """
+        int aFunction();
+        """
+        )
+
+        function_decl = self.tree_paths[0][1]
+        assert function_decl.cursor.kind == clang.CursorKind.FUNCTION_DECL
+        assert function_decl.cursor.spelling == "aFunction"
+        assert function_decl.cursor.result_type.kind == clang.TypeKind.INT
+
+    def test_function_decl_with_parameters(self):
+        self.__parse(
+            """
+        int aFunction(int firstParam, double secondParam);
+        """
+        )
+
+        function_decl = self.tree_paths[0][1]
+        assert function_decl.cursor.kind == clang.CursorKind.FUNCTION_DECL
+        assert function_decl.cursor.spelling == "aFunction"
+        assert function_decl.cursor.result_type.kind == clang.TypeKind.INT
+
+        function_decl_children = self.parser.get_parsed_infos_from_node_ids(
+            self.parser.get_node_ids_from_nodes(
+                self.parser.get_children_nodes_from_parent_parsed_info(function_decl)
+            )
+        )
```

Comment:
This looks gnarly. Helper function to wrap the 3 calls into a single call?

Reply:
Done.